src/visualizers/Differences.ts

```diff
@@ -55,25 +59,15 @@ class Differences extends P5Visualizer {
 
     first = 0
 
-    checkParameters() {
-        const status = super.checkParameters()
+    checkParameters(params: {[key: string]: unknown}) {
+        const status = super.checkParameters(params)
 
-        if (this.params.levels.value < 1) {
-            status.isValid = false
-            status.errors.push('Number of rows must be positive')
-        }
-        if (this.params.n.value < 0) {
-            status.isValid = false
-            status.errors.push(
-                "Number of entries in top row can't be negative"
-            )
-        }
-        if (this.params.n.value < this.params.levels.value) {
-            status.isValid = false
-            status.errors.push(
-                "Number of rows can't exceed length of first row"
-            )
-        }
+        if ((params.levels as number) < 1)
+            status.addError('Number of rows must be positive')
+        if ((params.n as number) < 0)
+            status.addError("Number of entries in top row can't be negative")
```

Comment:
These first two conditions should be moved into individual-parameter validates, correct?

Reply:
Yes they should, I wasn't sure how much we should be updating the visualizers to make optimal use of the new API, since our main focus has been on actually building the website rather than focusing on individual visualizers. We could try to make some changes, but to do so exhaustively for every visualizer would take quite some time (and understanding of the more complex ones).

Reply:
Ahh, that is a good point. The team discussed this situation and we agree that you should not have to dive into the guts of the visualizers. What we'd like to propose as a compromise is that I push one commit to this PR moving just these two specific checks into the respective parameter validates, for two reasons: (a) to have an example of the desired pattern clearly established in the code, and (b) to then make sure that everything works as desired with checks distributed into individual-parameter validates and comprehensive validate. Then we can leave all the other checks as you have them, and deal with it (likely ourselves) in a later PR. Does that sound reasonable? If so, let me know when is a good time for me to push a commit (or if your prefer to handle this localized refactor yourself). Thanks!